### PR TITLE
Fixes the issue with fn test and adds relevant test cases, fixes #650

### LINKED ISSUE #650: `fission fn test` breakage
```
soam@pf9 ~ [1]> cat ~/tmp/fail.js
something wrong
soam@pf9 ~> fission fn create --name fail --env nodejs --code ~/tmp/fail.js
function 'fail' created
soam@pf9 ~> fission fn test --name fail
Error calling function fail: 502 Failed to execute get logs request: Post /proxy/logs/fail: unsupported protocol scheme ""
```

### PR DESCRIPTION
**Pull request by vishal-biyani**
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/651)

### CODE REVIEW

File: test/tests/test_function_test/test_fn_test.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+log "Testing invalid function $valid_fn_name"
+fission fn test --name $invalid_fn_name > /tmp/invalid.log
+
+for i in {1..10}
+do
+    size=$(wc -c </tmp/invalid.log)
+    if [ $size == 0 ]
+    then
+        fission fn test --name $invalid_fn_name > /tmp/invalid.log
```
[the remark this answers is not in the document]
Comment (life1347) [reply test/tests/test_function_test/test_fn_test.sh]: LGTM except for one question ^
If you decide not to address it, you can merge it directly.

**Comment by life1347** [comment]
Sorry, @vishal-biyani . I accidentlly click the update branch and github merge master branch automatically...

File: test/tests/test_function_test/test_fn_test.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (vishal-biyani) [reply test/tests/test_function_test/test_fn_test.sh]: @life1347 I added a sleep but that did not help either. The issue is not that logs are coming late but that logs are not shipped before the container is deleted, IMO.

[the remark this answers is not in the document]
Comment (life1347) [reply test/tests/test_function_test/test_fn_test.sh]: Got it. I will merge it once the CI pass.

**Comment by life1347** [comment]
LGTM

**Pull request merged**